test/e2e/tests/multiple-transactions.spec.js

```diff
@@ -95,4 +101,125 @@ describe('Confirm transactions', function () {
+          text: 'Reject 2 transactions',
+          tag: 'a',
+        });
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+        await driver.waitForElementNotPresent('.loading-overlay__spinner');
+        // confirms first transaction
+        await driver.clickElement({ text: 'Confirm', tag: 'button' });
+
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindow(extension);
+        await driver.delay(regularDelayMs);
+        await driver.clickElement('[data-testid="home__activity-tab"]');
+
+        await driver.wait(async () => {
+          const confirmedTxes = await driver.findElements(
+            '.transaction-list__completed-transactions .transaction-list-item',
+          );
+          return confirmedTxes.length === 2;
+        }, 10000);
+      },
+    );
+  });
+
+  it('creates multiple queued transactions, then rejects', async function () {
+    await withFixtures(
+      {
+        dapp: true,
+        fixtures: new FixtureBuilder()
+          .withPermissionControllerConnectedToTestDapp()
+          .build(),
+        ganacheOptions,
+        title: this.test.title,
+      },
+      async ({ driver }) => {
+        await driver.navigate();
+        await driver.fill('#password', 'correct horse battery staple');
+        await driver.press('#password', driver.Key.ENTER);
+
+        // initiates a transaction from the dapp
+        await openDapp(driver);
+        // creates first transaction
+        await driver.clickElement({
+          text: 'Send EIP 1559 Transaction',
+          tag: 'button',
+        });
+        await driver.waitUntilXWindowHandles(3);
+        const windowHandles = await driver.getAllWindowHandles();
+        const extension = windowHandles[0];
+        const confirmation = windowHandles[2];
+        await driver.switchToWindowWithTitle('E2E Test Dapp', windowHandles);
+
+        // creates second transaction
+        await driver.clickElement({
+          text: 'Send EIP 1559 Transaction',
+          tag: 'button',
+        });
+        await driver.switchToWindow(confirmation);
+
+        // rejects second transaction
+        await driver.waitForSelector({
+          text: 'Reject 2 transactions',
+          tag: 'a',
+        });
+        await driver.clickElement({ text: 'Reject', tag: 'button' });
+        await driver.waitForElementNotPresent('.loading-overlay__spinner');
+        // rejects first transaction
+        await driver.clickElement({ text: 'Reject', tag: 'button' });
+
+        await driver.waitUntilXWindowHandles(2);
+        await driver.switchToWindow(extension);
+        await driver.delay(regularDelayMs);
+        await driver.clickElement('[data-testid="home__activity-tab"]');
+
+        const noTransactions = await driver.findElements(
+          '.transaction-list__empty-text',
+        );
+        assert.equal(
+          await noTransactions[0].getText(),
+          'You have no transactions',
```

Comment:
Is it safe to use text as a selector? Would it be more stable to verify there is no entries in the table?

Reply:
It's been used like that in other places, but I got to a better result asserting if `.transaction-list__empty-text` is present and visible and the list of transitions is not present.